pkg/providers/instancetype/suite_test.go

```diff
@@ -1186,6 +1185,22 @@ var _ = Describe("InstanceType Provider", func() {
 			vm := azureEnvNonZonal.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
 			Expect(vm.Zones).To(BeEmpty())
 		})
+		It("should support provisioning non-zonal instance types in zonal regions", func() {
+			nonZonalInstanceType := "Standard_NC6s_v3"
+			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+
+			pod := coretest.UnschedulablePod(coretest.PodOptions{
+				NodeSelector: map[string]string{v1.LabelInstanceTypeStable: nonZonalInstanceType},
+			})
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
```

Comment:
We seem to always repeat these steps should we combine them? In core this pattern makes more sense because we test more workload scheduling so we have a more diverse set of pods/workloads we pass in. Here almost every pod is just unschedulable pods with some node selector or no diversity at all.

Reply:
I think sometimes you want different orderings/things in between applied, provisioned, scheduled. I did simplify this test a bit based on other discussions with Alex.